nessie/src/main/java/org/apache/iceberg/nessie/NessieTableOperations.java

```diff
@@ -55,11 +55,25 @@ public class NessieTableOperations extends BaseMetastoreTableOperations {
    */
   public static final String NESSIE_COMMIT_ID_PROPERTY = "nessie.commit.id";
 
+  /**
+   * Name of the catalog property that controls whether metadata cleanup operations (e.g. GC) should
+   * be restricted for Nessie tables (defaults to {@code true}).
+   *
+   * <p>This is an advanced property. Setting it to {@code false} permits Iceberg code that is
+   * unaware of Nessie-specific behaviours (e.g. non-linear commit histories) to make metadata
+   * changes considering only the current branch history. It is a reasonable use case if there is
+   * only one branch defined on the Nessie Server side. If multiple Nessie branches are involved
+   * disabling this setting may lead to situations where some Nessie commits point to non-existent
+   * Iceberg metadata, which may break user expectations and/or make data in other Nessie branches
+   * unreadable.
+   */
+  public static final String RESTRICT_METADATA_CLEANUP_PROPERTY = "restrict-metadata-cleanup";
```

Comment:
so does that mean that users need to set the GC enabled and this flag to actually run GC? I'm not fully sure why we need both flags?

Reply:
is the issue that users can't override the gc flag because nessie always disables gc?

Reply:
> is the issue that users can't override the gc flag because nessie always disables gc?
Yes.

The new `RESTRICT_METADATA_CLEANUP_PROPERTY` (when set to `false`) instructs `NessieCatalog` to avoid modifying Iceberg GC properties. So the latter remain as set by the user.

Reply:
I'll see if we can clarify GC property management. WIP for now.